### PR TITLE
[Lock] remove deprecated Factory for LockFactory

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Branch?       | 5.0
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | none   
| License       | MIT
| Doc PR        | todo 

This remove the deprecated Factory and changes the alias in FrameworkBundle.

### CODE REVIEW

File: UPGRADE-5.0.md [hunk 1]
```diff
@@ -301,6 +301,11 @@ Intl
  * Removed `Intl::getLocaleBundle()`, use `Locales` instead
  * Removed `Intl::getRegionBundle()`, use `Countries` instead
 
+Lock
+----
+
+ * Removed `Factory`, use `LockFactory` instead
```
Comment (nicolas-grekas) [inline UPGRADE-5.0.md]: this should be submitted on 4.4, and UPGRADE-4.4.md should mention it too

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 2]
```diff
@@ -1485,12 +1485,12 @@ private function registerLockConfiguration(array $config, ContainerBuilder $cont
                 $container->setAlias('lock', new Alias('lock.'.$resourceName, false));
                 $container->setAlias(StoreInterface::class, new Alias('lock.store', false));
                 $container->setAlias(PersistStoreInterface::class, new Alias('lock.store', false));
-                $container->setAlias(Factory::class, new Alias('lock.factory', false));
+                $container->setAlias(LockFactory::class, new Alias('lock.factory', false));
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: the new alias should also exist on 4.4

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/lock.xml [hunk 3]
```diff
@@ -26,7 +26,7 @@
 
         <service id="lock.strategy.majority" class="Symfony\Component\Lock\Strategy\ConsensusStrategy" />
 
-        <service id="lock.factory.abstract" class="Symfony\Component\Lock\Factory" abstract="true">
+        <service id="lock.factory.abstract" class="Symfony\Component\Lock\LockFactory" abstract="true">
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/lock.xml]: that's for 4.4 too to me

File: UPGRADE-5.0.md (hunk 1 above)
Comment (Simperfit) [reply UPGRADE-5.0.md]: see https://github.com/symfony/symfony/pull/32508/files

File: src/Symfony/Component/Lock/LockFactory.php [hunk 4]
```diff
@@ -11,12 +11,42 @@
...
+use Psr\Log\NullLogger;
+
 /**
  * Factory provides method to create locks.
  *
  * @author Jérémy Derussé <jeremy@derusse.com>
  * @author Hamza Amrouche <hamza.simperfit@gmail.com>
  */
-class LockFactory extends Factory
+class LockFactory
```
Comment (xabbuh) [inline src/Symfony/Component/Lock/LockFactory.php]: Should it implement the `LoggerAwareInterface`?

Comment (Simperfit) [reply src/Symfony/Component/Lock/LockFactory.php]: yes! done

**Review by xabbuh — APPROVED** [review]

File: src/Symfony/Component/Lock/Tests/FactoryTest.php [hunk 5]
```diff
@@ -1,36 +0,0 @@
...
-use PHPUnit\Framework\TestCase;
-use Psr\Log\LoggerInterface;
-use Symfony\Component\Lock\Factory;
-use Symfony\Component\Lock\LockInterface;
-use Symfony\Component\Lock\StoreInterface;
-
-/**
- * @author Jérémy Derussé <jeremy@derusse.com>
- */
-class FactoryTest extends TestCase
```
Comment (jderusse) [inline src/Symfony/Component/Lock/Tests/FactoryTest.php]: should be reverted (and renamed into `LockFactoryTest`)

Comment (jderusse) [reply src/Symfony/Component/Lock/Tests/FactoryTest.php]: nvm

**Comment by Simperfit** [comment]
like @stof said, closing in favor of #32555  (this commits has been cherry-picked in the other PR)

**Pull request closed without merging**